pandas/tests/series/test_alter_axes.py

```diff
@@ -83,8 +83,9 @@ def test_rename_axis_supported(self):
         s = Series(range(5))
         s.rename({}, axis=0)
         s.rename({}, axis="index")
-        with pytest.raises(ValueError, match="No axis named 5"):
-            s.rename({}, axis=5)
+        # TODO: clean up shared index validation
```

Comment:
Commented this out for now as I didn't think it was hugely critical and seemed somewhat orthoganal to the test, but can add back in if there is a huge objection.

Ideally though we would have a centralized place for index validation that doesn't mutate the arguments of the caller.

Reply:
Fine to move this elsewhere. I do have a slight preference for not completely ignoring arguments that are there just for compatibility, but I don't know what our general rule is.

Reply:
Yea let me take another look I think should be able to keep this. Both `self._get_axis_number` and `self._get_axis_name` have some kind of validation in generic. 

I don't know if there is *one* consistent way we do this right now, but want to march towards that

Reply:
I _think_ the preference should be that the axis parameter should be validated if specified on a Series for consistency with most Series methods that accept axis (e.g. fillna) and *all* DataFrame methods that accept axis.

Reply:
> Both `self._get_axis_number` and `self._get_axis_name` have some kind of validation in generic.

normally you can just pass axis to super and it'll get validated? in this case there is also the call to self._set_name (cause of #28920). so probably will need the call to self._get_axis_number.